static/js/message_list.js

```diff
@@ -619,6 +619,20 @@ exports.MessageList.prototype = {
                 }
             }
         }, 0);
+    },
+
+    get_last_own_editable_message: function MessageList_get_last_own_editable_message() {
+        var msg_index = _.findLastIndex(this._items, {'sender_email': page_params.email});
```

Comment:
Can you do this check by `sender_id` instead?  That will be more robust in the event that the user's email address changes.